vispy/visuals/image.py

```diff
@@ -47,6 +47,8 @@ def __init__(self, data, method='subdivide', grid=(10, 10), **kwargs):
 
         self._texture = None
         self._interpolation = 'nearest'
+        if data.dtype == np.float64:
+            data = data.astype(np.float32)
```

Comment:
I think it would be better to do this conversion inside `set_data` Maybe we also need to first do something like `data = np.array(data, copy=False)` to convert whatever is given into a numpy array.